### ISSUE TITLE
More options in ModelTrainer

### ISSUE THREAD

**Issue by alanakbik** [flairNLP/flair#1252]
We sometimes get requests to add more options to the `ModelTrainer`. Let's collect them here so we can implement them in a PR:

- Annealing: Currently, `train_with_dev=True` makes the trainer use the dev data during training and anneal against training loss. However, we may want to separate these two behaviors: One variable to decide whether or not to add dev data to training, and another whether to anneal against loss or F1. This way, users could decide to evaluate against dev loss (see #1250).

- Batch-growth annealing needs more options (#1138) such as an upper bound on batch size, and a separate annealing rates for the learning rate and increase of batch sizes.

**Comment by lucaventurini** [comment]
Hi @alanakbik , thanks for addressing this.

Since it's on topic, I see that some months ago there was also an evaluation_metric option. May I ask why it was removed?

**Comment by krzysztoffiok** [comment]
Hi, i agree that it would be nice to have more metrics to choose from in this context. Especially for me it is interesting why people prefer F1 score and not MCC score since it is proven that the latter is simply more informative? (Chicco, Davide. "Ten quick tips for machine learning in computational biology." BioData mining 10.1 (2017): 35.)

**Comment by alanakbik** [comment]
We removed the metric option in the refactoring when we made the `ModelTrainer` agnostic to the task it trains. Since each task is evaluated differently, we added the evaluation to the interface of the `flair.nn.Model` class (i.e. `SequenceTagger` or `TextClassifier` implement their own evaluation routines). So each task now has a standard evaluation method that cannot be influenced by the `ModelTrainer`. This has the advantage that it simplifies the training method calls, but of course the problem that you can no longer choose an alternative evaluation metric. 

I'd like to find a solution to this that does not blow up our interfaces. So open to any suggestions :)

**Comment by lucaventurini** [comment]
Hi @alanakbik , I've finally had the time to read the paper mentioned in #1138 , and I see that it suggests to stop the batch growth when B has reached N/10, where N is the dataset size, and fallback to lr decay. Is this strategy already implemented?

I think it does not affect the stop parameters on lr and batch, one can simply set one of the two if using this strategy, but maybe it's something worth to keep in mind when the stopping criterion on batch size is implemented.

**Comment by alanakbik** [comment]
@lucaventurini ah cool, we currently don't have this implemented but this is something we should add!

**Comment by alanakbik** [comment]
This feature was added in #1138 !
